pages/document-upload/index.tsx

```diff
@@ -0,0 +1,270 @@
+import Wrapper from "@components/SDCWrapper";
+import DragAndDrop from "@components/DragAndDrop";
+// TODO session should be typed
+export default function documentUpload({ session }): JSX.Element {
+    const router = useRouter();
+    let { type } = router.query;
+
+    useEffect(() => {
+        if (
+            Object.keys(router.query).length === 0 &&
+            Object.getPrototypeOf(router.query) === Object.prototype
+        ) {
+            type = "other";
+        } else if (!("type" in router.query)) {
+            console.log("Unsupported query parameter. Redirected");
+            type = "other";
+            router.push("/document-upload").then(() => {
+                window.scrollTo({ top: 0 });
+            });
+        } else if (type === undefined) {
+            console.log(
+                "Query parameter not assigned. Upload will be sent to 'other'",
+            );
+            type = "other";
+        } else if (type === "") {
+            console.log(
+                "Query parameter not assigned. Upload will be sent to 'other'",
+            );
+            type = "other";
+            router.push("/document-upload").then(() => {
+                window.scrollTo({ top: 0 });
+            });
+        }
+    }, []);
+    // if no query param specified
+
+    const [uploadSuccess, setUploadSuccess] = useState(false);
+    const [isUploading, setIsUploading] = useState(false);
+    // allows future support of multiple file uploads
+    // if we really want to restrict to one document, remove the multiple from the input
+    const [files, setFiles] = useState<File[]>([]);
+    const upload = async () => {
+        setIsUploading(true);
+        const file = files[0];
+        try {
+            // TODO don't prefix file name, instead put random file name into database eventually
+            // TODO randomize filename
+            const res = await fetch(
+                `/api/upload/url?path=${type}&file=${session.user.email}-${session.id}-${file.name}`,
+            );
+            const data = await res.json();
+            const { url, fields } = data.data;
+            const formData = new FormData();
+
+            Object.entries({ ...fields, file }).forEach(([key, value]) => {
+                formData.append(key, value as string | Blob);
+            });
+            const fileUpload = await fetch(url, {
+                method: "POST",
+                body: formData,
+            });
+
+            if (fileUpload.ok) {
+                console.log("Uploaded successfully!");
+                setUploadSuccess(true);
+            } else {
+                // TODO
+                console.error("Upload failed.");
+                setUploadSuccess(false);
+            }
+        } catch (e) {
+            console.log(e);
+        }
+        setIsUploading(false);
+    };
+
+    const uploadDocumentUI = (): JSX.Element => {
+        return (
+            <Wrapper>
```

Comment:
We want to pass in the session into our wrapper or the navbar is not going to know if we're logged in
```suggestion
            <Wrapper session={session}>
```